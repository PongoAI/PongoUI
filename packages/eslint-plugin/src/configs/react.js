/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
