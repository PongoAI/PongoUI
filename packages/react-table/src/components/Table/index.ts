export * from './Table';
export * from './useTableStyles';
export * from './Table.types';
export * from './useTable';
