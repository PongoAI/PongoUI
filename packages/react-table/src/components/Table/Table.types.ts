import * as React from 'react';
import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type TableSlots = {
  /**
   * The root of the Table.
   */
  root: Slot<'div'>;
};

interface TableCommons {}

export interface TableProps extends ComponentProps<TableSlots>, TableCommons {}

export interface TableState extends ComponentState<TableSlots>, TableCommons {}
