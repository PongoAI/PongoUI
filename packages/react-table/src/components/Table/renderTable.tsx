import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { TableState, TableSlots } from './Table.types';

export const renderTable = (state: TableState) => {
  const { slots, slotProps } = getSlots<TableSlots>(state);

  return <slots.root {...slotProps.root} />;
};
