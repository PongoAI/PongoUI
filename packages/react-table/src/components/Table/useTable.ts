import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { TableProps, TableState } from './Table.types';

export const useTable = (props: TableProps, ref: React.Ref<HTMLElement>): TableState => {
  const { label } = props;

  const state: TableState = {
    label,
    components: {
      root: 'table',
    },
    root: getNativeElementProps('table', {
      ref,
      ...props,
    }),
  };

  state.root['aria-label'] = label;

  return state;
};
