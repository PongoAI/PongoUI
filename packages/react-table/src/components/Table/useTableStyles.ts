import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { TableState } from './Table.types';
import { createCustomFocusIndicatorStyle } from '@fluentui/react-tabster';
import { tokens } from '@pongo-ui/react-theme';

const useRootStyles = makeStyles({
  root: {},
});

export const useTableStyles = (state: TableState) => {
  const rootStyles = useRootStyles();

  state.root.className = mergeClasses(rootStyles.root, state.root.className);

  return state;
};
