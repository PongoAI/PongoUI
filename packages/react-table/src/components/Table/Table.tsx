import * as React from 'react';
import { useTable } from './useTable';
import { useTableStyles } from './useTableStyles';
import { renderTable } from './renderTable';
import type { TableProps } from './Table.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The Table component displays sets of data.
 */
export const Table: ForwardRefComponent<TableProps> = React.forwardRef((props, ref) => {
  const state = useTable(props, ref);

  useTableStyles(state);

  return renderTable(state);
});
Table.displayName = 'Table';
