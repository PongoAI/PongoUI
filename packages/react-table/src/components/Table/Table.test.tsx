import * as React from 'react';
import { Table } from './Table';
import { fireEvent, render, screen } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';
import { resetIdsForTests } from '@fluentui/react-utilities';

describe('Table', () => {
  isConformant({
    Component: Table,
    displayName: 'Table',
  });
});
