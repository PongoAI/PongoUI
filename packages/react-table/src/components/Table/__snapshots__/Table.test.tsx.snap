// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders a basic Table 1`] = `
<body>
  <div>
    <table
      aria-label="test"
      class="___ffvmmi0 f1w4nmp0 f1ddd56o f1ewtqcl f1wwrg80 fly5x3f fetgrfm fwsgm2r f1kc6foo f13jenf0 f16vktn6"
    />
  </div>
</body>
`;
